### ISSUE TITLE
Syntax highlighting of .tsx

### ISSUE THREAD

**Issue by shuhei** [shuhei/dotfiles#11]
Syntax highlighting hangs for .tsx with only vim-typescript. It works together with vim-jsx-typescript. Adding vim-jsx-pretty improves colors, but not as much as without vim-jsx-typescript.

yats.vim hangs too.

**Comment by shuhei** [comment]
`vim-jsx-typescript` sets `filetype=typescript.tsx` instead of `typescriptreact`...

**Comment by shuhei** [comment]
Fixed by setting `typescript.tsx` filetype for `.tsx` files.

Even though Vim sets `typescriptreact` filetype automatically to `.tsx` files, `ftplugin` of `typescript-vim` works with only `typescript` filetype (`typescript.tsx` is treated as both of `typescript` and `tsx`). `vim-jsx-typescript` sets up `ftdetect` to set `typescript.tsx` filetype for `.tsx` and that's why it made `typescript-vim` work.

After figuring out how it works, I set up the filetype detection in `.vimrc` and enabled only `vim-jsx-pretty` and `typescript-vim`, which have nice colors. https://github.com/shuhei/dotfiles/commit/0d0ea78377668196022bdf10ea0193f6e0877d20
